src/panels/lovelace/hui-root.ts

```diff
@@ -608,7 +608,18 @@ class HUIRoot extends LitElement {
       return;
     }
 
-    viewIndex = viewIndex === undefined ? 0 : viewIndex;
+    const isVisible = (view) =>
+      Boolean(
+        this._editMode ||
+          view.visible === undefined ||
+          (Array.isArray(view.visible) &&
+            view.visible.some((e) => e.user === this.hass!.user!.id)) ||
+          view.visible === true
```

Comment:
move this check up. It's super fast and so we should do it before iterating over an array